rchain/crypto.py

```diff
@@ -31,6 +31,16 @@ def verify(self, signature: bytes, data: bytes):
             signature, data, hashfunc=blake2b_32, sigdecode=sigdecode_der
         )
 
+    def verify_digest(self, signature: bytes, digest: bytes):
+        self._pub_key.verify_digest(signature, digest, sigdecode=sigdecode_der)
+
+    def to_hex(self, lower: bool = True) -> bytes:
+        hex_key = base64.b16encode(self.to_bytes())
+        return hex_key.lower() if lower else hex
+
+    def to_str(self, lower: bool = True) -> str:
+        return self.to_hex(lower).decode('utf8')
```

Comment:
With the above change this could become `self.to_hex().upper()`, but I'd not introduce `to_str()`, `to_hex()` is enough.